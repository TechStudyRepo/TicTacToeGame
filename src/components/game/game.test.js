import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import Game from './game';
import PlayerDetails from './player-details';

configure({ adapter: new Adapter() });

describe('Result Component', () => {
    const initialState = {
        registerUser: {
            user: {
                user1: "P1",
                user2: "P2"
            }
        },
        player: 'P1'
    },
        mockStore = configureStore();
    let store;

    test("renders", () => {
        store = mockStore(initialState);
        const wrapper = shallow(<Provider store={store}>
            <Game />
        </Provider>);
        expect(wrapper.exists()).toBe(true);
    });

    test('should render search somponent', () => {
        store = mockStore(initialState);
        const wrapper = mount(<Provider store={store}>
            <Game />
        </Provider>);
        expect(wrapper.children(PlayerDetails).length).toEqual(0);
    });

    test('Test Snapshot', () => {
        store = mockStore(initialState);
        const wrapper = shallow(<Provider store={store}>
            <Game />
        </Provider>);
        expect(wrapper).toMatchSnapshot();
    });

})
