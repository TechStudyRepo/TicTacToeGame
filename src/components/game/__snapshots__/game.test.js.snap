// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Component Test Snapshot 1`] = `ShallowWrapper {}`;
